sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationHeadMockTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using Azure.Core.Pipeline;
+using Azure.Core.Testing;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Http;
+using NUnit.Framework;
+using System;
+
+namespace Azure.Data.AppConfiguration.Tests
+{
+    [TestFixture(true)]
+    [TestFixture(false)]
+    public partial class ConfigurationMockTests: ClientTestBase
+    {
+        class HeadRequestPolicy : SynchronousHttpPipelinePolicy
+        {
+            public readonly static HttpPipelinePolicy Shared = new HeadRequestPolicy();
+
+            private HeadRequestPolicy() { }
+
+            public override void OnSendingRequest(HttpPipelineMessage message)
+                => message.Request.Method = RequestMethod.Head;          
```

Comment:
Shouldn't this only change if the very is already a `GET`?

Reply:
It will change only if the verb is not HEAD :-)

But more seriously, I think it's not necessary as checking the value of the verb might be expensive, and might happen multiple times; once in the caller and once in the callee.

Reply:
Since adding this policy creates an instance of a client that only sends HEAD requests, are we concerned about the possibility that a user could send a HEAD request to an endpoint that doesn’t support it?

Reply:
Yeah, we would have to think what to do here. Maybe we would just make sure the errors are reasonable.

Reply:
Would it even be an error?  If you change `DELETE <endpoint>` into `HEAD <endpoint>` it's just going to return the properties.  _Maybe_ you'll notice if the status code comes back as 200 instead of 202?  I really think you should only change `GET` into `HEAD` if we're going with this approach.